main.go

```diff
@@ -213,7 +291,42 @@ func validateHostPort(hostPort string) error {
 	return nil
 }
 
-func newEtcdClient(machines []string, certFile, keyFile, caFile, username, password string) (etcd.KeysAPI, error) {
+func newEtcdV3Client(machines []string, tlsCert, tlsKey, tlsCACert string) (*etcdv3.Client, error) {
+	etcdCfg := etcdv3.Config{
+		Endpoints: machines,
+		TLS: newTLSConfig(tlsCert, tlsKey, tlsCACert),
+	}
+	cli, err := etcdv3.New(etcdCfg)
+	if err != nil {
+		return nil, err
+	}
+	return cli, nil
+}
+
+func newTLSConfig(tlsCertFile, tlsKeyFile, tlsCACertFile string) *tls.Config {
```

Comment:
you can just use newHTTPSTransport and extract the `TLSConfig` from there, no reason to duplicate this here.